### PR TITLE
fix: Using table owned dal operator instead of get from query context

### LINKED ISSUE #8022: flake test: 02_0001_create_table_with_external_location test failure in cluster mode
**Summary**

https://github.com/datafuselabs/databend/actions/runs/3174130129/jobs/5170683099

Can give some advice? CC: @ariesdevil

### PR DESCRIPTION
**Pull request by ariesdevil**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

1. Create OpenDAL operator from table info and use this operator do other things.
2. Using QueryCtx::get_storage_operator() iff table info include no connection infos.

I think this PR will fix #8022 , cc @zhang2014 

Fixes #8022

### CODE REVIEW

File: src/query/storages/util/src/file_meta_data_reader.rs [hunk 1]
```diff
@@ -40,6 +41,7 @@ impl FileMetaDataReader {
 impl Loader<FileMetaData> for Arc<dyn TableContext> {
     async fn load(
         &self,
+        _op: Operator,
```
Comment (Xuanwo) [inline src/query/storages/util/src/file_meta_data_reader.rs:44]: This op is not used?

Comment (ariesdevil) [reply src/query/storages/util/src/file_meta_data_reader.rs:44]: Yup, just for satisfying the trait fn.

Comment (Xuanwo) [reply src/query/storages/util/src/file_meta_data_reader.rs:44]: Seems weird. Maybe we can always use the input op and let the caller decide which op to use?

Comment (ariesdevil) [reply src/query/storages/util/src/file_meta_data_reader.rs]: fixed:)

**Comment by ariesdevil** [comment]
@Xuanwo not ready, some codes need to refactor, will be soon.

File: src/query/service/tests/it/storages/fuse/operations/navigate.rs [hunk 2]
```diff
@@ -84,7 +84,12 @@ async fn test_fuse_navigate() -> Result<()> {
     assert_eq!(second_snapshot, loc);
     let version = TableMetaLocationGenerator::snapshot_version(loc.as_str());
     let snapshots: Vec<_> = reader
-        .snapshot_history(loc, version, fuse_table.meta_location_generator().clone())
+        .snapshot_history(
+            fuse_table.get_operator(),
```
Comment (Xuanwo) [inline src/query/service/tests/it/storages/fuse/operations/navigate.rs:88]: The same question. Can we init the reader with op?

Comment (ariesdevil) [reply src/query/service/tests/it/storages/fuse/operations/navigate.rs:88]: I'm working on this.

**Review by Xuanwo — APPROVED** [review]
Nice patch!

**Review by zhyass — APPROVED** [review]
LGTM

**Review by dantengsky — APPROVED** [review]

**Pull request merged**